progs/jubjub.py

```diff
@@ -219,3 +220,29 @@ def __str__(self):
 
     async def double(self):
         return self
+
+
+async def share_mul(context: Mpc, p: Point, x: list) -> SharedPoint:
+    """
+    The multiplication of the share of a field element and a point
+    e.g. [x]G -> [X], where G is a point on the given elliptic curve
+    x is the bitwise shared value,
+    starting from the most significant bit.
+
+    Source: Section `Scalar multiplication` of
+    https://andrea.corbellini.name/2015/05/17/elliptic-curve-cryptography-a-gentle-introduction/
+    """
+    if isinstance(p, Ideal):
+        return SharedIdeal(Jubjub())
+
+    product = Ideal(p.curve)
+    addend = p
+    x.reverse()
+    one_share = context.Share(1)
+
+    for i_ in x:
+        if await (await equality(context, i_, one_share)).open():
```

Comment:
We should make `equality` be able to take in an `int` or `GFElement` and have it create a share to compare with. This is fine for now.

Reply:
We can make this more concurrent--
```suggestion
equalities = await asyncio.gather(*[equality(context, i_, one_share) for i_ in x])
opened_equalities = await asyncio.gather(*[eq.open() for eq in equalities])

for eq in opened_equalities:
  if eq:
    product += addend
  
  addend = addend.double()

return await SharedPoint.from_point(context, product)
```